test/store_test.dart

```diff
@@ -0,0 +1,157 @@
+        async.elapse(new Duration(milliseconds: 20));
+        expect(count, equals(1));
+        async.elapse(new Duration(milliseconds: 30));
+        expect(count, equals(2));
+        // Make sure that there aren't any queued up.
+        async.elapse(new Duration(milliseconds: 500));
+        expect(count, equals(2));
+      });
+    });
+
+    test('should trigger in response to an action', () {
+      store.triggerOnAction(action);
+      store.listen(expectAsync((Store listenedStore) {
+        expect(listenedStore, equals(store));
+      }));
+      action();
+    });
+
+    test(
+        'should execute a given method and then trigger in response to an action',
+        () async {
+      bool wasTriggered = false;
+      void onAction(Null _) {
+        wasTriggered = true;
+      }
+      store.triggerOnAction(action, onAction);
+      store.listen(expectAsync((Store listenedStore) {
+        expect(listenedStore, equals(store));
+        expect(wasTriggered, isTrue);
+      }));
+      action();
+    });
+
+    test(
+        'should execute a given method and then trigger in response to a conditional action',
+        () {
+      bool wasTriggered = false;
+      bool onAction(Null _) {
+        wasTriggered = true;
+        return true;
+      }
+      store.triggerOnConditionalAction(action, onAction);
+      store.listen(expectAsync((Store listenedStore) {
+        expect(listenedStore, equals(store));
+        expect(wasTriggered, isTrue);
+      }));
+      action();
+    });
+
+    test(
+        'should execute a given method but NOT trigger in response to a conditional action',
+        () {
+      new FakeAsync().run((FakeAsync async) {
+        bool onAction(Null _) => false;
+        store.triggerOnConditionalAction(action, onAction);
+        store.listen((Store listenedStore) {
+          fail('Event should not have been triggered');
+        });
+        action();
+        async.flushMicrotasks();
+      });
+    });
+
+    test(
+        'should execute a given async method and then trigger in response to an action',
+        () {
+      bool afterTimer = false;
+      Future<Null> asyncCallback(Null _) async {
+        await new Future<Null>.delayed(new Duration(milliseconds: 30));
+        afterTimer = true;
+      }
+      store.triggerOnAction(action, asyncCallback);
+      store.listen(expectAsync((Store listenedStore) {
+        expect(listenedStore, equals(store));
+        expect(afterTimer, isTrue);
+      }));
+      action();
+    });
+
+    test(
+        'should execute a given method and then trigger in response to an action with payload',
+        () {
+      final Action<int> _action = new Action<int>();
+      int counter = 0;
+      store.triggerOnAction(_action, (int payload) => counter = payload);
+      store.listen(expectAsync((Store listenedStore) {
+        expect(listenedStore, equals(store));
+        expect(counter, equals(17));
+      }));
+      return _action(17);
+    });
+  });
+}
```

Comment:
would be nice to have some flutter tests too to check the integration works right